linera-service/src/linera/main.rs

```diff
@@ -517,59 +485,47 @@ impl Runnable for Job {
+                        error!("{}", e);
+                        faulty_validators
+                            .entry((name, address))
+                            .or_default()
+                            .push(e);
                     }
-                    let query = linera_core::data_types::ChainInfoQuery::new(chain_id);
-                    match node.handle_chain_info_query(query).await {
-                        Ok(response) => {
-                            info!(
-                                "Validator {name:?} at {address} sees chain {chain_id} at block height {} and epoch {:?}",
-                                response.info.next_block_height,
-                                response.info.epoch,
-                            );
-                            if response.check(name).is_ok() {
-                                info!("Signature for public key {name} is OK.");
-                                num_ok_validators += 1;
-                            } else {
-                                error!("Signature for public key {name} is NOT OK.");
-                                faulty_validators.push((name, address, format!("{:#?}", response)));
-                            }
-                        }
-                        Err(e) => {
-                            error!("Failed to get chain info for validator {name:?} at {address} and chain {chain_id}:\n{e}");
-                            faulty_validators.push((
-                                name,
-                                address,
-                                "Failed to get chain info.".to_string(),
-                            ));
-                            continue;
-                        }
+                    if let Err(e) = context
+                        .check_matching_network_description(address, &node)
+                        .await
+                    {
+                        error!("{}", e);
+                        faulty_validators
+                            .entry((name, address))
+                            .or_default()
+                            .push(e);
+                    }
+                    if let Err(e) = context
+                        .check_validator_chain_info_response(None, address, &node, chain_id)
+                        .await
+                    {
+                        error!("{}", e);
+                        faulty_validators
+                            .entry((name, address))
+                            .or_default()
+                            .push(e);
                     }
                 }
+                let num_ok_validators = committee.validators().len() - faulty_validators.len();
                 if !faulty_validators.is_empty() {
                     println!("{:#?}", faulty_validators);
                 }
-                println!("{}/{} OK.", num_ok_validators, committee.validators().len());
```

Comment:
@christos-h Are we parsing this somewhere? Is it ok to turn it into `info!`?

Reply:
context: We generally print only machine-parsable outputs on `stdout` (preferably in JSON or newline-separated JSON)